Dockerfile

```diff
@@ -104,97 +112,102 @@ RUN sudo ln -s ../lib/node_modules/npm/bin/npm-cli.js /usr/local/bin/npm
-    git clone https://github.com/beacon-biosignals/ray ${RAY_ROOT} && \
-    git -C ${RAY_ROOT} checkout ${RAY_COMMIT} && \
-    mkdir -p ${JLL_JULIA_PROJECT}/deps && \
-    ln -s ${RAY_ROOT} ${JLL_JULIA_PROJECT}/deps/ray && \
-    cd ${JLL_JULIA_PROJECT}/deps/ray && \
+    git clone https://github.com/beacon-biosignals/ray ${RAY_REPO} && \
+    git -C ${RAY_REPO} checkout ${RAY_COMMIT} && \
+    #
+    # Build using the final Ray.jl destination
+    mkdir -p ${RAY_JLL_PROJECT}/deps && \
+    ln -s ${RAY_REPO} ${RAY_JLL_PROJECT}/deps/ray && \
+    cd ${RAY_JLL_PROJECT}/deps/ray && \
     #
     # Allow builders to clear just the Ray CLI cache.
     if [ "${RAY_CACHE_CLEAR}" != "false" ]; then \
         bazel clean --expunge && \
-        rm -rf ${RAY_GEN_CACHE_DIR}/*; \
+        rm -rf ${RAY_REPO_CACHE}/*; \
     fi && \
     #
     # The Ray `BUILD.bazel` includes a bunch of `copy_to_workspace` rules which copy build output
     # into the Ray worktree. When we only restore the Bazel cache then re-building causes these
     # rules to be skipped resulting in `error: [Errno 2] No such file or directory`. By manually
     # saving/restoring these files we can work around this.
-    if [ -d ${RAY_GEN_CACHE_DIR}/.git ]; then \
-        cd ${RAY_GEN_CACHE_DIR} && \
+    if [ -d ${RAY_REPO_CACHE}/.git ]; then \
+        cd ${RAY_REPO_CACHE} && \
         cp -rp --parents \
             python/ray/_raylet.so \
             python/ray/core/generated \
             python/ray/serve/generated \
             python/ray/core/src/ray/raylet/raylet \
             python/ray/core/src/ray/gcs \
-            ${RAY_ROOT} && \
+            ${RAY_REPO} && \
         cd -; \
     fi && \
     #
     # Build the dashboard
-    cd ${JLL_JULIA_PROJECT}/deps/ray/python/ray/dashboard/client && \
+    cd ${RAY_JLL_PROJECT}/deps/ray/python/ray/dashboard/client && \
     npm ci && \
     npm run build && \
     #
     # Build Ray for Python
-    cd ${JLL_JULIA_PROJECT}/deps/ray/python && \
+    cd ${RAY_JLL_PROJECT}/deps/ray/python && \
     pip install --verbose . && \
     #
     # By copying the entire Ray worktree we can easily restore missing files without having to
     # delete the cache and build from scratch.
-    mkdir -p ${RAY_GEN_CACHE_DIR} && \
-    cp -rfp ${RAY_ROOT}/. ${RAY_GEN_CACHE_DIR} && \
-    #
-    # Remove directory to avoid conflict with a future COPY
-    rm -rf ${JLL_JULIA_PROJECT}
+    mkdir -p ${RAY_REPO_CACHE} && \
+    cp -rfp ${RAY_REPO}/. ${RAY_REPO_CACHE}
 
 # Copy over artifacts generated during the previous stages
 COPY --chown=ray --link --from=deps ${HOME}/.julia ${HOME}/.julia
 
 # Setup ray_julia_jll
-ARG JLL_ROOT=/ray_julia_jll
-COPY --chown=ray ray_julia_jll ${JLL_ROOT}
-RUN --mount=type=cache,sharing=locked,target=/mnt/bazel-cache,uid=1000,gid=100 \
+ARG RAY_JLL_REPO=${HOME}/.julia/dev/ray_julia_jll
+COPY --chown=ray ray_julia_jll ${RAY_JLL_REPO}
+RUN --mount=type=cache,sharing=locked,target=${BAZEL_CACHE},uid=${UID},gid=${GID} \
```

Comment:
this is gonna change with #158 too right?

Reply:
Yup, will need to update all the paths here